docs/developers/react-development-guidelines.md

```diff
@@ -0,0 +1,402 @@
+---
+id: react-development-guidelines
+title: React Development Guidelines
+---
+
+[comment]: # (mx-abstract)
+
+## Introduction
```

Comment:
please keep in mind that Docusaurus is different than MD files for handling headings. In the current format, `Introduction` is heading 1, and `Abstracting the logic away into hooks and function` will be a sub-chapter of the `Introduction` in the navigation panel on the right side. 
![Screenshot 2023-12-05 at 14 21 09](https://github.com/multiversx/mx-docs/assets/51945539/7b58f5df-ab13-412b-96f1-6b46bf3a8868)

So use `##` for heading 1, `###` for heading 2 and so on